extensions/interactions/NumberWithUnits/directives/NumberWithUnits.js

```diff
@@ -0,0 +1,145 @@
+              $scope.NumberWithUnitsForm.answer.$setValidity(
+                FORM_ERROR_TYPE, true);
+            } catch (parsingError) {
+              errorMessage = parsingError.message;
+              $scope.NumberWithUnitsForm.answer.$setValidity(
+                FORM_ERROR_TYPE, false);
+            }
+          });
+
+          $scope.submitAnswer = function(answer) {
+            try {
+              var numberWithUnits =
+                NumberWithUnitsObjectFactory.fromRawInputString(answer);
+              $scope.onSubmit({
+                answer: numberWithUnits,
+                rulesService: numberWithUnitsRulesService
+              });
+            } catch (parsingError) {
+              errorMessage = parsingError.message;
+              $scope.NumberWithUnitsForm.answer.$setValidity(
+                FORM_ERROR_TYPE, false);
+            }
+          };
+
+          $scope.$on(EVENT_PROGRESS_NAV_SUBMITTED, function() {
+            $scope.submitAnswer($scope.answer);
+          });
+
+          $scope.isAnswerValid = function() {
+            return (!$scope.NumberWithUnitsForm.$invalid &&
+              $scope.answer !== '');
+          };
+
+          $scope.$watch(function() {
+            return $scope.answer;
+          }, function() {
+            $scope.setAnswerValidity({
+              answerValidity: $scope.isAnswerValid()
+            });
+          });
+        }
+      ]
+    };
+  }
+]);
+
+oppia.directive('oppiaResponseNumberWithUnits', [
+  'NumberWithUnitsObjectFactory', 'HtmlEscaperService',
+  'UrlInterpolationService', function(NumberWithUnitsObjectFactory,
+      HtmlEscaperService, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/NumberWithUnits/directives/' +
+        'number_with_units_directive.html'),
+      controller: ['$scope', '$attrs', function($scope, $attrs) {
+        var answer = HtmlEscaperService.escapedJsonToObj($attrs.answer);
+        $scope.answer = NumberWithUnitsObjectFactory.fromDict(
+          answer).toString();
+      }]
+    };
+  }
+]);
+
+oppia.directive('oppiaShortResponseNumberWithUnits', [
+  'NumberWithUnitsObjectFactory', 'HtmlEscaperService',
+  'UrlInterpolationService', function(NumberWithUnitsObjectFactory,
+      HtmlEscaperService, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/NumberWithUnits/directives/' +
+        'number_with_units_directive.html'),
+      controller: ['$scope', '$attrs', function($scope, $attrs) {
+        var answer = HtmlEscaperService.escapedJsonToObj($attrs.answer);
+        $scope.answer = NumberWithUnitsObjectFactory.fromDict(
+          answer).toString();
+      }]
+    };
+  }
+]);
+
+// Rule evaluation for number with units (will be implemented in M1.3).
```

Comment:
Is there anything else beside this in M1.3? If not then I suggest that we merge milestones and do it in single PR.

Reply:
My bad. Lets not do this here. Let's keep it separate. Also note that @seanlip suggested that we enable the NumberWithUnits interaction in next milestone so that it is available in production in July releas.

Reply:
Ok, no problem. There are lots of files to be added in M1.3 for validation and corresponding spec files. We can enable it but till then currency units will not be introduced. What to do in this case?

Reply:
Yep. We are going to release priliminary version without currency units.